Tests/SwiftParserTest/translated/InitDeinitTests.swift

```diff
@@ -438,4 +438,361 @@ final class InitDeinitTests: XCTestCase {
+          fixIts: ["remove redundant 'async'"]
+        ),
+      ],
+      fixedSource: """
+        class FooClassDeinitializerA {
+          deinit async {}
+        }
+        """
+    )
+  }
+
+  func testDeinitNameAsyncThrows() {
+    assertParse(
+      """
+      class FooClassDeinitializerA {
+        deinit 1️⃣x async 2️⃣throws {}
+      }
+      """,
+      diagnostics: [
+        DiagnosticSpec(locationMarker: "1️⃣", message: "deinitializers cannot have a name", fixIts: ["remove 'x'"]),
+        DiagnosticSpec(locationMarker: "2️⃣", message: "deinitializers cannot throw", fixIts: ["remove 'throws'"]),
+      ],
+      fixedSource: """
+        class FooClassDeinitializerA {
+          deinit async {}
+        }
+        """
+    )
+  }
+
+  func testDeinitParamsAsyncThrows() {
+    assertParse(
+      """
+      class FooClassDeinitializerA {
+        deinit1️⃣() async 2️⃣throws {}
+      }
+      """,
+      diagnostics: [
+        DiagnosticSpec(locationMarker: "1️⃣", message: "deinitializers cannot have parameters", fixIts: ["remove parameter clause"]),
+        DiagnosticSpec(locationMarker: "2️⃣", message: "deinitializers cannot throw", fixIts: ["remove 'throws'"]),
+      ],
+      fixedSource: """
+        class FooClassDeinitializerA {
+          deinit async {}
+        }
+        """
+    )
+  }
+
+  func testDeinitNameParamsThrowsAsync() {
+    assertParse(
+      """
+      class FooClassDeinitializerA {
+        deinit 1️⃣x2️⃣() 3️⃣throws async {}
+      }
+      """,
+      diagnostics: [
+        DiagnosticSpec(locationMarker: "1️⃣", message: "deinitializers cannot have a name", fixIts: ["remove 'x'"]),
+        DiagnosticSpec(locationMarker: "2️⃣", message: "deinitializers cannot have parameters", fixIts: ["remove parameter clause"]),
+        DiagnosticSpec(locationMarker: "3️⃣", message: "deinitializers cannot throw", fixIts: ["remove 'throws'"]),
+      ],
+      fixedSource: """
+        class FooClassDeinitializerA {
+          deinit async {}
+        }
+        """
+    )
+  }
+
+  func testDeinitNameParamsAwaitThrows() {
+    assertParse(
+      """
+      class FooClassDeinitializerA {
+        deinit 1️⃣x2️⃣() 3️⃣await 4️⃣throws {}
+      }
+      """,
+      diagnostics: [
+        DiagnosticSpec(locationMarker: "1️⃣", message: "deinitializers cannot have a name", fixIts: ["remove 'x'"]),
+        DiagnosticSpec(locationMarker: "2️⃣", message: "deinitializers cannot have parameters", fixIts: ["remove parameter clause"]),
+        DiagnosticSpec(locationMarker: "3️⃣", message: "expected async specifier; did you mean 'async'?", fixIts: ["replace 'await' with 'async'"]),
+        DiagnosticSpec(locationMarker: "4️⃣", message: "deinitializers cannot throw", fixIts: ["remove 'throws'"]),
+      ],
+      fixedSource: """
+        class FooClassDeinitializerA {
+          deinit async {}
+        }
+        """
+    )
+  }
+
+  func testAsyncDeinit() {
+    // TODO: Investigate why this produces no diagnostics
```

Comment:
This is expected for now. `async` is parsed as a modifier like `public` because you can have an `async var x: Int`.

Reply:
Updated comment